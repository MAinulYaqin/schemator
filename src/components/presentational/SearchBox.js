import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import "typeface-roboto";

const Input = styled.input`
  width: 100%;
  background: #2b2b2b;
  border: 1px solid #444;
  outline: none;
  color: white;
  font-family: Roboto;
  font-size: 12px;
  padding: 8px;
  box-sizing: border-box;
`;

const SearchBox = ({ className, value, onChange }) => (
  <Input className={className} value={value} onChange={onChange} />
);

SearchBox.propTypes = {
  className: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func
};

export default SearchBox;
