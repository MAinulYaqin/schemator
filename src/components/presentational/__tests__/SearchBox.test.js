import React from "react";
import { shallow } from "enzyme";

import SearchBox from "../SearchBox";

describe("SearchBox", () => {
  const setup = propOverrides => {
    const props = {
      value: "Laravel",
      onChange: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<SearchBox {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should calls 'onChange' while changed", () => {
    const { wrapper, props } = setup();

    wrapper.simulate("change");

    expect(props.onChange).toBeCalled();
  });
});
