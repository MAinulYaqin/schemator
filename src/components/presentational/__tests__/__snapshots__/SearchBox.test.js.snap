// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox should renders default 1`] = `
<styled.input
  onChange={[Function]}
  value="Laravel"
/>
`;
